test/factories/azure.rb

```diff
@@ -1,9 +1,8 @@
 FactoryBot.define do
-    factory :azure_cr, :parent => :compute_resource, :class => ForemanAzureRM::AzureRM do
-      provider 'AzureRM'
-      user 'azurermuser'
-      password 'azurermpassword'
-      url 'http://azurerm.example.com'
-      uuid 'azurermuuid'
-    end
+  factory :azure_rm, :parent => :compute_resource, :class => ForemanAzureRM::AzureRM do
+    add_attribute(:user) { 'azurermuser' }
+    add_attribute(:password) { 'azurermpassword' }
+    add_attribute(:url) { 'http://azurerm.example.com' }
+    add_attribute(:uuid) { 'azurermuuid' }
```

Comment:
It would be much helpful to use the mock data format as using in actually:

i.e.  `add_attribute(:uuid) { 'a0a086e2-da7a-4059-aafe-9cc07cc519cd' }` 

same for user, password, url.